src/ai/backend/manager/api/gql/error.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+import strawberry
+
+if TYPE_CHECKING:
+    from ai.backend.common.exception import BackendAIError as BackendAIErrorException
+    from ai.backend.common.exception import ErrorCode as ErrorCodeData
+
+
+@strawberry.type(
+    name="ErrorCode",
+    description="Structured error code representing domain, operation and error detail",
+)
+class ErrorCode:
+    domain: str = strawberry.field(description="Error domain (e.g., 'api', 'kernel', 'session')")
+    operation: str = strawberry.field(
+        description="Operation where error occurred (e.g., 'create', 'delete', 'update')"
+    )
+    error_detail: str = strawberry.field(
+        description="Specific error detail (e.g., 'not-found', 'timeout', 'forbidden')"
+    )
+
+    @classmethod
+    def from_dataclass(cls, error_code: ErrorCodeData) -> ErrorCode:
+        """
+        Create a GraphQL ErrorCode from an ErrorCode dataclass.
+        """
+        return cls(
+            domain=error_code.domain,
+            operation=error_code.operation,
+            error_detail=error_code.error_detail,
+        )
+
+
+@strawberry.type(
+    name="BackendAIError", description="Structured error information from Backend.AI API"
+)
+class BackendAIError:
```

Comment:
It seems we can specify a name, so the class name itself should be taken with a different name.